packages/arb-token-bridge-ui/src/util/withdrawals/fetchWithdrawals.ts

```diff
@@ -129,36 +105,33 @@ export const fetchWithdrawals = async ({
   const currentPageStart = pageNumber * pageSize
   const currentPageEnd = currentPageStart + pageSize
 
-  const partialEthWithdrawalsFromEventLogs = [...ethWithdrawalsFromEventLogs]
-    // event logs start from the earliest, we need to reverse them
-    .reverse()
-    .slice(currentPageStart, currentPageEnd)
-  const partialTokenWithdrawalsFromEventLogs = [
-    ...tokenWithdrawalsFromEventLogs
+  // we need timestamps to sort token withdrawals along ETH withdrawals
+  const tokenWithdrawalsFromEventLogsWithTimestamp =
+    await attachTimestampToTokenWithdrawals({
+      withdrawals: tokenWithdrawalsFromEventLogs,
+      l2Provider
+    })
+
+  // get ETH and token withdrawals that will be displayed on the current page
+  const partialWithdrawalsFromEventLogs = [
```

Comment:
NIT: Maybe outside the scope, but I would rename this to `paginatedWithdrawalsFromEventLogs` because `partialWithdrawalsFromEventLogs` gives me the impression that the funds are partially withdrawn :)

Reply:
I like that